app/src/main/java/ch/eureka/eurekapp/ui/meeting/MeetingNavigationScreen.kt

```diff
@@ -0,0 +1,225 @@
+        } else if (uiState.meeting == null || uiState.errorMsg != null) {
+          ErrorScreen(message = uiState.errorMsg ?: stringResource(R.string.meeting_not_found))
+        } else {
+          MapContent(modifier = Modifier.padding(padding), uiState = uiState)
+        }
+      })
+}
+
+/** Loading indicator screen. */
+@Composable
+private fun LoadingScreen() {
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(16.dp)
+              .testTag(MeetingNavigationScreenTestTags.LOADING_INDICATOR),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
+        CircularProgressIndicator()
+        Spacer(modifier = Modifier.height(16.dp))
+        Text(
+            text = stringResource(R.string.loading_meeting_location),
+            style = MaterialTheme.typography.bodyMedium)
+      }
+}
+
+/**
+ * Error message screen.
+ *
+ * @param message The error message to display to the user.
+ */
+@Composable
+private fun ErrorScreen(message: String) {
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(16.dp)
+              .testTag(MeetingNavigationScreenTestTags.ERROR_MESSAGE),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
+        Text(
+            text = message,
+            style = MaterialTheme.typography.bodyLarge,
+            color = MaterialTheme.colorScheme.error)
+      }
+}
+
+/**
+ * Main map content displaying the meeting location.
+ *
+ * @param modifier Modifier to be applied to the root composable.
+ * @param uiState The current UI state.
+ */
+@Composable
+private fun MapContent(modifier: Modifier = Modifier, uiState: MeetingNavigationUIState) {
+  val meeting = uiState.meeting ?: return
+  val meetingLocation = meeting.location?.let { LatLng(it.latitude, it.longitude) } ?: return
+
+  Column(modifier = Modifier.fillMaxSize().then(modifier)) {
+    // Google Map
+    Box(modifier = Modifier.weight(1f).fillMaxWidth()) {
+      MapView(meetingLocation = meetingLocation, meetingName = meeting.location.name)
+    }
+
+    // Information Card
+    InfoCard(
+        modifier = Modifier.fillMaxWidth().padding(16.dp), locationName = meeting.location.name)
+  }
+}
+
+/**
+ * Google Map view showing meeting location.
+ *
+ * @param meetingLocation The meeting location coordinates.
+ * @param meetingName The name of the meeting location.
+ */
+@Composable
+private fun MapView(meetingLocation: LatLng, meetingName: String) {
+  val cameraPositionState = rememberCameraPositionState {
+    position = CameraPosition.fromLatLngZoom(meetingLocation, Constants.DEFAULT_MAP_ZOOM)
+  }
+
+  GoogleMap(
+      modifier = Modifier.fillMaxSize().testTag(MeetingNavigationScreenTestTags.GOOGLE_MAP),
+      cameraPositionState = cameraPositionState) {
+        // Meeting location marker
+        Marker(
+            state = MarkerState(position = meetingLocation),
```

Comment:
Important : Here it would be better to use a `rememBerMarkerState` so that the`MarkerState` object is preserved across recompositions. It will improve the performance because `GoogleMap` composable can be recomposed multiple time per seconds.
```suggestion
            state = rememberMarkerState(position = meetingLocation),
```